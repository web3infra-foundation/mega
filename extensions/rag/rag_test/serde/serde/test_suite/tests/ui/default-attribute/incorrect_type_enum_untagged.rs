// Tests that type error points to the path in attribute

use serde_derive::Deserialize;

#[derive(Deserialize)]
#[serde(untagged)]
enum Enum {
    // Newtype variants do not use the provided path, so it is forbidden here
    // Newtype(#[serde(default = "main")] u8),
    Tuple(u8, #[serde(default = "main")] i8),
    Struct {
        #[serde(default = "main")]
        f1: u8,
        f2: u8,
        #[serde(default = "main")]
        f3: i8,
    },
}

fn main() {}
