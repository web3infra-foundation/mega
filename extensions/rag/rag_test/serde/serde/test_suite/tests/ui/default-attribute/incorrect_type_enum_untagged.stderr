error[E0308]: `match` arms have incompatible types
  --> tests/ui/default-attribute/incorrect_type_enum_untagged.rs:10:33
   |
5  | #[derive(Deserialize)]
   |          -----------
   |          |
   |          this is found to be of type `i8`
   |          `match` arms have incompatible types
...
10 |     Tuple(u8, #[serde(default = "main")] i8),
   |                                 ^^^^^^ expected `i8`, found `()`

error[E0308]: `match` arms have incompatible types
  --> tests/ui/default-attribute/incorrect_type_enum_untagged.rs:12:27
   |
5  | #[derive(Deserialize)]
   |          -----------
   |          |
   |          this is found to be of type `u8`
   |          `match` arms have incompatible types
...
12 |         #[serde(default = "main")]
   |                           ^^^^^^ expected `u8`, found `()`

error[E0308]: `match` arms have incompatible types
  --> tests/ui/default-attribute/incorrect_type_enum_untagged.rs:15:27
   |
5  | #[derive(Deserialize)]
   |          -----------
   |          |
   |          this is found to be of type `i8`
   |          `match` arms have incompatible types
...
15 |         #[serde(default = "main")]
   |                           ^^^^^^ expected `i8`, found `()`
